src/movies.js

```diff
@@ -1,35 +1,246 @@
+//       genre: ['Crime', 'Drama'],
+//       score: 9.3
+//     }
+//   ])
+// );
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(arr) {
+  if (arr.length) {
+    let scores = arr.map((element) => element.score);
+    console.log(scores);
+    let filteredScores = Array.from(scores, (item) => item || 0);
+    console.log(filteredScores);
+    let average = filteredScores.reduce((acc, value) => acc + value, 0);
+    return parseFloat((average / filteredScores.length).toFixed(2));
+  } else {
+    return 0;
+  }
+}
+
+// console.log(`The score avg of an empty array is 0: ` + scoresAverage([]));
+
+// console.log(
+//   `The score avg is ` +
+//     scoresAverage([
+//       {
+//         title: 'The Shawshank Redemption',
+//         year: 1994,
+//         director: 'Frank Darabont',
+//         duration: '2h 22min',
+//         genre: ['Crime', 'Drama'],
+//         score: ''
+//       },
+//       {
+//         title: 'The Shawshank Redemption',
+//         year: 1994,
+//         director: 'Frank Darabont',
+//         duration: '2h 22min',
+//         genre: ['Crime', 'Drama'],
+//         score: 9.7
+//       }
+//     ])
+// );
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(arr) {
+  let dramas = arr.filter((element) => element.genre.includes('Drama'));
+  return scoresAverage(dramas);
+}
+
+// console.log(
+//   dramaMoviesScore([
+//     {
+//       title: 'The Shawshank Redemption',
+//       year: 1994,
+//       director: 'Frank Darabont',
+//       duration: '2h 22min',
+//       genre: ['Crime'],
+//       score: 9.3
+//     },
+//     {
+//       title: 'The Shawshank Redemption',
+//       year: 1994,
+//       director: 'Frank Darabont',
+//       duration: '2h 22min',
+//       genre: ['Crime', 'Drama'],
+//       score: 9.7
+//     }
+//   ])
+//);
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+// function orderByYear(arr) {
+//   let copiedArr = [...arr]; // using spread to not affect original array
+//   let sortedArr = copiedArr.sort(function (a, b) {
+//     return a.year - b.year;
+//   });
+//   return sortedArr;
+// }
+
+// helping function from the internet:
+// yourArray.sort(function(a, b) {
+//   var adate = /* ...parse the date in a.date... */,
+//       bdate = /* ...parse the date in b.date... */,
+//       rv = adate - bdate;
+//   if (rv === 0) {
+//       rv = a.title.localeCompare(b.title);
+//   }
+//   return rv;
+// });
+
+function orderByYear(arr) {
+  let copiedArr = [...arr]; // using spread to not affect original array
+  let sortedArr = copiedArr.sort(function (a, b) {
+    let adate = a.year;
+    let bdate = b.year;
+    let rv = adate - bdate;
+    if (rv === 0) {
+      rv = a.title.localeCompare(b.title);
+    }
+    return rv;
+  });
+  return sortedArr;
+}
```

Comment:
This solution is correct but you could do it with the same logic but a bit shorter:
```
function orderByYear(movies) {
  const ordered = [...movies].sort((a, b) => {
    if (a.year !== b.year) {
      return a.year - b.year;
    }
    return a.title.localeCompare(b.title);
  })
  return ordered;
}
```

Reply:
smart